### PR TITLE
Update docs for filter command

### LINKED ISSUE #127: [Week 11] 2: Update user docs
https://nus-cs2103-ay2324s1.github.io/website/schedule/week11/project.html#2-update-user-docs

Updated draft of the UG & landing page.

### PR DESCRIPTION
**Pull request by bwangpj**
Update documentation for filter command.

Part of #127.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -54,6 +54,18 @@ Display all your stored contacts.
 
 ---
 
+### Filtering Contacts by Tag: `filter`
+
+Search and display contacts with tag matching the given keyword (case-insensitive).
```
Comment (Cloud7050) [inline docs/UserGuide.md:59]: Is it an exact tag match?

Comment (bwangpj) [reply docs/UserGuide.md:59]: Yep exact match ignoring case

**Review by Cloud7050 — APPROVED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (Cloud7050) [reply docs/UserGuide.md:59]: It works with multiple keywords to match multi-word tags. Maybe worth using plural in the docs for the param value?

Comment (bwangpj) [reply docs/UserGuide.md:59]: Ok will do!

**Review by rayshawntan — APPROVED** [review]

**Pull request merged**